### PR TITLE
feat: Infer Model Name automatically if empty in Model Forms

### PR DESCRIPTION
**Pull request by blessedcoolant**
## What type of PR is this? (check all applicable)

- [x] Feature

## Have you discussed this change with the InvokeAI team?
- [x] No
      
## Description

Automatically infer the name of the model from the path supplied IF the model name slot is empty. If the model name is not empty, we presume that the user has entered a model name or made changes to it and we do not touch it in order to not override user changes.

## Related Tickets & Documents

- Addresses: #4443

### CODE REVIEW

File: invokeai/frontend/web/src/features/ui/components/tabs/ModelManager/subpanels/AddModelsPanel/AdvancedAddDiffusers.tsx [hunk 1]
```diff
@@ -92,6 +92,14 @@ export default function AdvancedAddDiffusers(props: AdvancedAddDiffusersProps) {
           label="Model Location"
           placeholder="Provide the path to a local folder where your Diffusers Model is stored"
           {...advancedAddDiffusersForm.getInputProps('path')}
+          onBlur={(e) => {
+            if (advancedAddDiffusersForm.values['model_name'] === '') {
+              advancedAddDiffusersForm.setFieldValue(
+                'model_name',
+                e.currentTarget.value.split('\\').splice(-1)[0] as string
```
Comment (psychedelicious) [inline invokeai/frontend/web/src/features/ui/components/tabs/ModelManager/subpanels/AddModelsPanel/AdvancedAddDiffusers.tsx:99]: This won't work for Linux or macOS.

Also, this is unsafe... should assign the potential name to a variable and check if it is a strong first before setting.

File: invokeai/frontend/web/src/features/ui/components/tabs/ModelManager/subpanels/AddModelsPanel/AdvancedAddCheckpoint.tsx [hunk 2]
```diff
@@ -100,6 +100,17 @@ export default function AdvancedAddCheckpoint(
           label="Model Location"
           required
           {...advancedAddCheckpointForm.getInputProps('path')}
+          onBlur={(e) => {
```
Comment (psychedelicious) [inline invokeai/frontend/web/src/features/ui/components/tabs/ModelManager/subpanels/AddModelsPanel/AdvancedAddCheckpoint.tsx:103]: Same comment

**Review by psychedelicious — CHANGES_REQUESTED** [review]

File: invokeai/frontend/web/src/features/ui/components/tabs/ModelManager/subpanels/AddModelsPanel/AdvancedAddDiffusers.tsx (hunk 1 above)
Comment (blessedcoolant) [reply invokeai/frontend/web/src/features/ui/components/tabs/ModelManager/subpanels/AddModelsPanel/AdvancedAddDiffusers.tsx:99]: Why doesn't it work on Linux or macOS?

Comment (blessedcoolant) [reply invokeai/frontend/web/src/features/ui/components/tabs/ModelManager/subpanels/AddModelsPanel/AdvancedAddDiffusers.tsx]: Added check for UNIX paths.

Comment (blessedcoolant) [reply invokeai/frontend/web/src/features/ui/components/tabs/ModelManager/subpanels/AddModelsPanel/AdvancedAddDiffusers.tsx]: Actually might need to do some more stronger validation. I'll do it in a bit.

Comment (blessedcoolant) [reply invokeai/frontend/web/src/features/ui/components/tabs/ModelManager/subpanels/AddModelsPanel/AdvancedAddDiffusers.tsx]: Updated. Should now work for folders and checkpoints across both windows and unix paths. Tested on Windows. No obvious issues. Let me know if it works on Linux / macOS.

**Review by psychedelicious — APPROVED** [review]

**Pull request merged**